guidance/database/tags.md

```diff
@@ -0,0 +1,62 @@
+# Database tags guidance
```

Comment:
I like the level of detail here, but I think we should wait to merge until the `data_conventions.yaml` becomes an .md file. I will try (?) to get that done today, assuming I can find the time!